examples/src/test/kotlin/AggregatesBuilderTest.kt

```diff
@@ -0,0 +1,939 @@
+        assertEquals("Name1", resultsFlow.toList()[0].name)
+        assertEquals(4, resultsFlow.toList()[0].socialNetwork.size)
+    }
+
+    @Test
+    fun graphLookupOptionsTest() = runBlocking {
+        data class Results(val name: String, val friends: List<String>, val hobbies: List<String>, val socialNetwork: List<Users>)
+        val resultsFlow = contactsCollection.aggregate<Results>(listOf(
+                // :snippet-start: graph-lookup-options
+                Aggregates.graphLookup(
+                    "contacts",
+                    "\$${Users::friends.name}", Users::friends.name, Users::name.name, Results::socialNetwork.name,
+                    GraphLookupOptions().maxDepth(1).restrictSearchWithMatch(
+                        Filters.eq(Users::hobbies.name, "golf")
+                    )
+                )
+                // :snippet-end:
+            ))
+        assertEquals("Name1", resultsFlow.toList()[0].name)
+        assertEquals(2, resultsFlow.toList()[0].socialNetwork.size)
+    }
+
+    @Test
+    fun sortByCount() = runBlocking {
+        data class Results(@BsonId val id: String, val count: Int)
+
+        val resultsFlow = movieCollection.aggregate<Results>(listOf(Aggregates.unwind("\$${Movie::genres.name}"),
+            // :snippet-start: sort-by-count
+            Aggregates.sortByCount("\$${Movie::genres.name}"),
+            // :snippet-end:
+            Aggregates.sort(Sorts.descending(Results::count.name, "_id"))))
+        val results = resultsFlow.toList()
+        val actual = listOf(Results("Drama", 8), Results("Crime", 3), Results("Action", 2), Results("Thriller", 1), Results("Sci-Fi", 1), Results("Romance", 1), Results("Mystery", 1),)
+        assertEquals(results, actual)
+    }
+
+    @Test
+    fun replaceRootTest() = runBlocking {
+        // :snippet-start: replace-root-data-class
+        data class Libro(val titulo: String)
+        data class Book(val title: String, val spanishTranslation: Libro)
+        // :snippet-end:
+        data class Results(val titulo: String)
+        val translateCollection = database.getCollection<Book>("books_translate")
+        val books = listOf(Book("Movie1", Libro("Libro1")), Book("Book2", Libro("Libro2")), Book("Movie3", Libro("Libro3")))
+        translateCollection.insertMany(books)
+
+        val resultsFlow = translateCollection.aggregate<Results>(listOf(
+            // :snippet-start: replace-root
+            replaceRoot("\$${Book::spanishTranslation.name}")
+            //  :snippet-end:
+
+        ))
+        assertEquals("Libro1", resultsFlow.toList()[0].titulo)
+        translateCollection.drop()
+    }
+
+    @Test
+    fun addFieldsTest() = runBlocking {
+        data class Results (val cast: List<String>, val type: String)
+        val resultsFlow = movieCollection.aggregate<Results>(listOf(
+            // :snippet-start: add-fields
+            addFields(Field(
+                Results::cast.name, listOf("Nicholas Cage")),
```

Comment:
modest proposal to only use Nicholas Cage movies when doing examples with the mflix data  set?